File: pawtracker-backend/src/main/java/com/pawtracker/entities/Requests/ColonyReportRequest.java
package com.pawtracker.entities.Requests;

import lombok.AllArgsConstructor;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

@Data
@AllArgsConstructor
public class ColonyReportRequest {
    private UUID colonyId;
    private String title;
    private List<CatReportEntryRequest> catsFed;
    private List<CatReportEntryRequest> catsMissing;
    private UUID userId;
    private LocalDateTime datetime;
}

File: PawTrackerColonies-frontend/src/models/ColonyReportRequest.ts
import { CatReportEntryRequest } from "./CatReportEntryRequest";

export interface ColonyReportRequest {
  colonyId: string;              
  title: string;
  catsFed: CatReportEntryRequest[];
  catsMissing: CatReportEntryRequest[];
  userId: string;                
  datetime: string;             
}